packages/account/src/providers/transaction-response/getAllDecodedLogs.ts

```diff
@@ -0,0 +1,78 @@
+import type { JsonAbi } from '@fuel-ts/abi-coder';
+import { Interface, BigNumberCoder } from '@fuel-ts/abi-coder';
+import { ZeroBytes32 } from '@fuel-ts/address/configs';
+import { ReceiptType } from '@fuel-ts/transactions';
+
+import type {
+  TransactionResultCallReceipt,
+  TransactionResultReceipt,
+} from './transaction-response';
+
+export interface DecodedLogs<T = unknown> {
+  logs: T[];
+  groupedLogs: Record<string, T[]>;
+}
+
+/**
+ * @hidden
+ */
+export function getAllDecodedLogs<T = unknown>(opts: {
+  receipts: Array<TransactionResultReceipt>;
+  mainAbi: JsonAbi;
+  externalAbis?: Record<string, JsonAbi>;
+}): DecodedLogs<T> {
+  const { receipts, mainAbi, externalAbis = {} } = opts;
+
+  /**
+   * This helper decodes logs from transaction receipts.
+   * It loops through all receipts and decodes two types of logs:
+   * - ReceiptType.LogData
+   * - ReceiptType.Log
+   *
+   * The "mainAbi" parameter represents the ABI of the main contract used to create the transaction
+   * or the ABI from a script used within a "BaseInvocationScope" context.
+   * The "externalAbis" parameter is a record of contract ABIs that are also part of the transaction.
+   * These ABIs were added using `contract.addContracts` or through a multi-call with `contract.multiCall`.
+   *
+   * @param receipts - The array of transaction result receipts.
+   * @param mainAbi - The ABI of the script or main contract.
+   * @param externalAbis - The record of external contract ABIs.
+   * @returns An array of decoded logs from Sway projects.
+   */
```

Comment:
What's the problem with having the docs above the method?

```suggestion
/**
 * @hidden
 * This helper decodes logs from transaction receipts.
 *
 * It loops through all receipts and decodes two types of logs:
 * - ReceiptType.LogData
 * - ReceiptType.Log
 *
 * The "mainAbi" parameter represents the ABI of the main contract used to create the transaction
 * or the ABI from a script used within a "BaseInvocationScope" context.
 * The "externalAbis" parameter is a record of contract ABIs that are also part of the transaction.
 * These ABIs were added using `contract.addContracts` or through a multi-call with `contract.multiCall`.
 *
 * @param receipts - The array of transaction result receipts.
 * @param mainAbi - The ABI of the script or main contract.
 * @param externalAbis - The record of external contract ABIs.
 * @returns An array of decoded logs from Sway projects.
 */
export function getAllDecodedLogs<T = unknown>(opts: {
  receipts: Array<TransactionResultReceipt>;
  mainAbi: JsonAbi;
  externalAbis?: Record<string, JsonAbi>;
}): DecodedLogs<T> {
  const { receipts, mainAbi, externalAbis = {} } = opts;
```

Reply:
faa8c4d6fae0a9b511944d123233ad5f3cda9056